compiler/src/dmd/backend/cod3.d

```diff
@@ -1694,8 +1695,18 @@ void doswitch(ref CodeBuilder cdb, block *b)
 
         // Put into casevals[0..ncases] so we can sort then slice
         assert(ncases < size_t.max / (2 * CaseVal.sizeof));
-        CaseVal *casevals = cast(CaseVal *)malloc(ncases * CaseVal.sizeof);
-        assert(casevals);
+
+        CaseVal[10] tmp = void;
+        CaseVal[] casevals;
+        if (ncases <= tmp.length)
+            casevals = tmp[0 .. ncases]; // allocate on stack
+        else
+        {
+            auto p = cast(CaseVal *)malloc(ncases * CaseVal.sizeof);
+            assert(p);
+            casevals = p[0 .. ncases];
+        }
```

Comment:
This stack allocation logic used to be copy pasted all over dmd, but it's now available using a neat abstraction:
```suggestion
        import dmd.common.string : SmallBuffer;
        auto sb = SmallBuffer!(CaseVal)(ncases, tmp[]);
        CaseVal[] casevals = sb[];
```

Reply:
ok